src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -603,7 +590,14 @@ public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
                 try
                 {
                     bool hasCompleteReferences = true;
+
                     var newReferences = new List<MetadataReference>();
+
+                    // we don't consider metadata reference being actually exist on disk as a part of complete reference
+                    // since we know some projects do have those but many just work fine since those dlls are not actually used.
```

Comment:
one of reason I didn't do that was not sure where to put that check. surely it can't be here since file check is too big perf hit for this hot path.